core/src/main/scala/kafka/Kafka.scala

```diff
@@ -65,11 +65,12 @@ object Kafka extends Logging {
 
   private def buildServer(props: Properties): Server = {
     val config = KafkaConfig.fromProps(props, false)
-    if (config.processRoles.isEmpty) {
+    if (config.requiresZookeeper) {
```

Comment:
What if we call `processRoles.isDefined` here?

Reply:
Hmm `requiresZookeeper` seemed more explicit. Using `processRoles` seemed a little more obscure.